File_Adapter/FileAdapter.cs

```diff
@@ -33,22 +33,22 @@ public partial class FileAdapter : BHoMAdapter
         /**** Properties                                ****/
         /***************************************************/
 
-
         /***************************************************/
         /**** Constructors                              ****/
         /***************************************************/
+        [Input("folder", "Defaults to the path of your default drive (usually C://)")]
         [Input("fileName","Insert filename with extension.\nCurrently supports only .json and .bson file types.")]
-        public FileAdapter(string folder = "C:\\", string fileName = "objects.json")
+        public FileAdapter(string folder = null, string fileName = "objects.json")
         {
+            if (folder == null)
+                Path.GetPathRoot(Environment.SystemDirectory);
+
             if (folder.Count() > 2 && folder.ElementAt(1) != ':')
             {
-                folder = @"C:\Users\" + Environment.UserName + @"\AppData\Roaming\BHoM\DataSets\" + folder;
+                folder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), @"\BHoM\DataSets\", folder);
```

Comment:
This fails to me. You can test:
```c#
void Main()
{
	string folder = "Materials";
	string path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), @"\BHoM\DataSets\", folder);
	string path2 = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "BHoM", "DataSets", folder);
	Console.WriteLine(path);
	Console.WriteLine(Directory.Exists(path));
	
	Console.WriteLine("---");
	Console.WriteLine(path2);
	Console.WriteLine(Directory.Exists(path2));
	
}
```
Returns:
```
\BHoM\DataSets\Materials
False
---
C:\Users\epignatel\AppData\Roaming\BHoM\DataSets\Materials
True
```

Reply:
Sorry about this! I didn't think of testing because was wrapped in an unused condition. Changed now.